examples/typescript/swap.ts

```diff
@@ -0,0 +1,275 @@
+/**
+ * Example to demonstrate creating and adding to liquidity pools, swapping between two fungible asset.
+ *
+ * Steps:
+ * 1. Create two accounts (Alice and Bob)
+ * 2. Fund Alice and Bob
+ * 3. Create a FA with Alice as the owner
+ * 4. Create another FA with Bob as the owner
+ * 5. Create a liquidity pool with Alice's FA and Bob's FA
+ * 6. Swap between Alice's FA and Bob's FA
+ */
+
+import {
+  Account,
+  AccountAddress,
+  Aptos,
+  Bool,
+  Ed25519PrivateKey,
+  Hex,
+  HexInput,
+  MoveObject,
+  MoveString,
+  U64,
+  ViewRequestData,
+} from "aptos";
+
+const readline = require("readline").createInterface({
+  input: process.stdin,
+  output: process.stdout,
+});
+
+const getOptimalLpAmount = async (
+  aptos: Aptos,
+  swap: AccountAddress,
+  token1Addr: HexInput,
+  token2Addr: HexInput,
+): Promise<void> => {
+  const payload: ViewRequestData = {
+    function: `0x${swap.toStringWithoutPrefix()}::router::optimal_liquidity_amounts`,
+    arguments: [
+      `0x${Hex.fromHexInput(token1Addr).toStringWithoutPrefix()}`,
+      `0x${Hex.fromHexInput(token2Addr).toStringWithoutPrefix()}`,
+      false,
+      "200000",
+      "300000",
+      "200",
+      "300",
```

Comment:
Are these U64s?

I suspect one of the things we should probably fix on the API side for View functions is the encoding, and allow BCS encoding of inputs, and use the ABI for translation.

Reply:
+1